pkg/lightning/backend/local/engine.go

```diff
@@ -1393,9 +1402,8 @@ func (sw *sstWriter) writeKVs(kvs []common.KvPair) error {
 	internalKey := sstable.InternalKey{
 		Trailer: uint64(sstable.InternalKeyKindSet),
 	}
-	var lastKey []byte
 	for _, p := range kvs {
-		if bytes.Equal(p.Key, lastKey) {
+		if sw.lastKey != nil && bytes.Equal(p.Key, sw.lastKey) {
```

Comment:
why we cannot use local `lastKey` as before, what's changed?

Reply:
maybe add some comment to explain why, as we don't need it in previous version

Reply:
`writeKVs` is allowed to be called more than one times. However, in one SST file all keys should be sorted and deduped. So we can't use a local variable in `writeKVs`.